### ISSUE TITLE
Unity IAP 4.12 bug with Google Play Games plugins and Game Play Services - GRADLE/PLUGINS issue

### ISSUE THREAD

**Issue by ohyouknow24** [playgameservices/play-games-plugin-for-unity#3322]
This form is for reporting Unity Plugin issues only. To report an issue with the Play Games Service (non-SDK related), check [Google Play Games Services Support](https://developers.google.com/games/services/support).
*Once you've read this section and determined that your issue is appropriate for this repository, please delete this section.*

**Describe the bug**
Game Play Services will not work with Unity 4.12 IAP, only 4.11 and down. This forces you to not use the plugins from the resolver, but instead use gradles for a successful build, however, successfully building with gradles and IAP 4.12 then won't allow game play services to work forces you to pick between either functioning IAP or functioning game play services by removing IAP and building out with the plugins in the android plugin folder. 

**To Reproduce**
1. Download plugin 10.14, Download EDM4U 1.283, Use any version of unity
2. Install 4.12 IAP package
3. Install game play services and make sure everything is correct
4. Keep all plugins in the folder
5. Build game

**Expected behavior**
To build out the game so that IAP and game play services work, the same way it does with 4.11

**Observed behavior**
It throws duplicate classes errors and launcher duplicate errors. However, if you don't use the plugins, but instead opt for the custom Gradle, it will resolve and remove the plugins in build out, however, then your game play services will not work

**Versions**
- Unity version: 6
- Google Play Games Plugin for Unity version: 10.14

**Additional context**
I’m hoping someone can help me. I’m using Unity 6, with EDM4U 1.283, with Google Play Plugin 10.14.

This issue was nonexistent using Unity IAP 4.11, however, due to needing the 6.0 Android Library, I have no choice but to upgrade.

Here’s what’s happening. Thus far, I was using the plugins that were inside of the Assets/plugins/android folder. I could force resolve and it would build out the individual plugins I needed and using IAP 4.11, I could successfully build out my game with everything working (game play services included).

However, when trying to make a build using unity 6, with IAP 4.12, if I try to make a build using the plugins in the folder, it always fails building because it says there are duplicates (even though there are not, I have checked every folder). If I remove those plugins, I can build successfully, however, then my game play services achievements will no longer function.

Using the custom gradles doesn’t solve the problem either. I can build successfully, and even in the logcat it shows that it is trying to authentic game play services, and it comes back true, and yet, it has not signed in the player nor does the achievement button work as it shows “not implemented.” Again, this is with the Gradle.

So my dilemma is this. In order to adhere to googles 6.0 library, I have to decide between removing my IAP so that I can use the plugins to build out my game with game play services working, or I have to remove game play services so that IAP continue working.

Is there any workarounds anyone knows of? I have tried and looked up everything possible.

Am I missing something in my Gradle? I will post my Gradle below if there is something I am supposed to implement in the decencies that would enable game play services, please let me know:

custom main Gradle:
`apply plugin: 'com.android.library'
apply from: '../shared/keepUnitySymbols.gradle'
**APPLY_PLUGINS**

dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    implementation 'androidx.localbroadcastmanager:localbroadcastmanager:1.0.0'
// Android Resolver Dependencies Start
    implementation 'com.google.games:gpgs-plugin-support:0.10.14' // Assets/GooglePlayGames/Editor/GooglePlayGamesPluginDependencies.xml:11
// Android Resolver Dependencies End
**DEPS**}

// Android Resolver Exclusions Start
android {
  packaging {
      exclude ('/lib/armeabi/*' + '*')
      exclude ('/lib/mips/*' + '*')
      exclude ('/lib/mips64/*' + '*')
      exclude ('/lib/x86/*' + '*')
      exclude ('/lib/x86_64/*' + '*')
  }
}
// Android Resolver Exclusions End
android {
    namespace "com.unity3d.player"
    ndkPath "**NDKPATH**"
    ndkVersion "**NDKVERSION**"

    compileSdk **APIVERSION**
    buildToolsVersion = "**BUILDTOOLS**"

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_17
        targetCompatibility JavaVersion.VERSION_17
    }

    defaultConfig {
        minSdk **MINSDK**
        targetSdk **TARGETSDK**
        ndk {
            abiFilters **ABIFILTERS**
            debugSymbolLevel **DEBUGSYMBOLLEVEL**
        }
        versionCode **VERSIONCODE**
        versionName '**VERSIONNAME**'
        consumerProguardFiles 'proguard-unity.txt'**USER_PROGUARD**
**DEFAULT_CONFIG_SETUP**
    }

    lint {
        abortOnError false
    }

    androidResources {
        noCompress = **BUILTIN_NOCOMPRESS** + unityStreamingAssets.tokenize(', ')
        ignoreAssetsPattern = "!.svn:!.git:!.ds_store:!*.scc:!CVS:!thumbs.db:!picasa.ini:!*~"
    }**PACKAGING**
}
**IL_CPP_BUILD_SETUP**
**SOURCE_BUILD_SETUP**
**EXTERNAL_SOURCES**`

build.gradle:
`apply plugin: 'com.android.library'
apply from: '../shared/keepUnitySymbols.gradle'

dependencies {
    implementation fileTree(dir: 'libs', include: ['*.jar'])
    implementation 'androidx.localbroadcastmanager:localbroadcastmanager:1.0.0'
// Android Resolver Dependencies Start
    implementation 'com.google.games:gpgs-plugin-support:0.10.14' // Assets/GooglePlayGames/Editor/GooglePlayGamesPluginDependencies.xml:11
// Android Resolver Dependencies End
    implementation(name: 'facebook-android-sdk-4.17.0', ext:'aar')
    implementation(name: 'facebook-android-wrapper-7.9.0', ext:'aar')
    implementation(name: 'appcompat-v7-23.4.0', ext:'aar')
    implementation(name: 'animated-vector-drawable-23.4.0', ext:'aar')
    implementation(name: 'UnityAds', ext:'aar')
    implementation(name: 'common', ext:'aar')
    implementation(name: 'cardview-v7-23.4.0', ext:'aar')
    implementation(name: 'UnityAndroidPermissions', ext:'aar')
    implementation(name: 'customtabs-23.4.0', ext:'aar')
    implementation(name: 'support-vector-drawable-23.4.0', ext:'aar')
    implementation project(':unityLibrary:GooglePlayGamesManifest.androidlib')

}

// Android Resolver Exclusions Start
android {
  packaging {
      exclude ('/lib/armeabi/*' + '*')
      exclude ('/lib/mips/*' + '*')
      exclude ('/lib/mips64/*' + '*')
      exclude ('/lib/x86/*' + '*')
      exclude ('/lib/x86_64/*' + '*')
  }
}
// Android Resolver Exclusions End
android {
    namespace "com.unity3d.player"
    ndkPath "/Applications/Unity/Hub/Editor/6000.0.23f1/PlaybackEngines/AndroidPlayer/NDK"
    ndkVersion "23.1.7779620"

    compileSdk 35
    buildToolsVersion = "34.0.0"

    compileOptions {
        sourceCompatibility JavaVersion.VERSION_17
        targetCompatibility JavaVersion.VERSION_17
    }

    defaultConfig {
        minSdk 26
        targetSdk 35
        ndk {
            abiFilters 'armeabi-v7a', 'arm64-v8a'
            debugSymbolLevel 'none'
        }
        versionCode 44
        versionName '1.8.2'
        consumerProguardFiles 'proguard-unity.txt'

    }

    lint {
        abortOnError false
    }

    androidResources {
        noCompress = ['.unity3d', '.ress', '.resource', '.obb', '.bundle', '.unityexp'] + unityStreamingAssets.tokenize(', ')
        ignoreAssetsPattern = "!.svn:!.git:!.ds_store:!*.scc:!CVS:!thumbs.db:!picasa.ini:!*~"
    }

    packaging {
        jniLibs {
                useLegacyPackaging true
        }
    }
}

// Dependencies for "com.unity.purchasing". This section is automatically generated.
afterEvaluate {
    dependencies {
        implementation 'com.android.billingclient:billing:6.2.1'
    }
}
// End of dependencies for "com.unity.purchasing".`

**Comment by poorva-google** [comment]
Could you please upgrade to the latest plugin 11.0.1 ?

**Comment by adamvision** [comment]
Was there ever any resolution to this? I cannot get Unity IAP, Google Play Services Plugin to work together without constant gradle issues when trying to build even the simplest of solutions. I am using the latest plugin of both, have tried in various unity versions from 2022.3 - current.

**Comment by xxluky** [comment]
I can confirm. Unity 6000.0.25
I can build a project without GPGS without a problem. As soon as I import the 0.11.1v or 2.0v I get several errors:

1. UnityPlayerActivity.java uses or overrides a deprecated API.
2. > Could not resolve all files for configuration ':launcher:releaseRuntimeClasspath'.  
    > Failed to transform classes.jar (project :unityLibrary) ...
3. conflict with com.google.games:gpgs-plugin-support:2.0.0 - No variants of com.google.games:gpgs-plugin-support:2.0.0 match the consumer attributes:
4. UnityPlayerActivity.java uses or overrides a deprecated API
5. Gradle build failed

So in short, this plugin needs an update of everything to work in my project. This sucks. I was using this plugin for years an my whole project is dependent. Now I am not able to build my game!! Very frustrating!

**Comment by xxluky** [comment]
> I can confirm. Unity 6000.0.25 I can build a project without GPGS without a problem. As soon as I import the 0.11.1v or 2.0v I get several errors:
> 
> 1. UnityPlayerActivity.java uses or overrides a deprecated API.
> 2. > Could not resolve all files for configuration ':launcher:releaseRuntimeClasspath'.
>    > Failed to transform classes.jar (project :unityLibrary) ...
> 3. conflict with com.google.games:gpgs-plugin-support:2.0.0 - No variants of com.google.games:gpgs-plugin-support:2.0.0 match the consumer attributes:
> 4. UnityPlayerActivity.java uses or overrides a deprecated API
> 5. Gradle build failed
> 
> So in short, this plugin needs an update of everything to work in my project. This sucks. I was using this plugin for years an my whole project is dependent. Now I am not able to build my game!! Very frustrating!

Sorry, here is my update: After 4 days of trying this is what helped:

1. Add this to the main manifest:
`
      <property
        android:name="android.adservices.AD_SERVICES_CONFIG"
        android:resource="@xml/gma_ad_services_config"
        tools:replace="android:resource" />`
2. I deleted all Library related assets. For me it was Firebase, IAP, AdMob. After I deleted these plugins the purpose of this was to delete Android and iOS folder in Plugins in order to re-generate all depedency files in these folders.
3. Import GPGS v2.0.0 with their Dependency Manager 1.2.182. (1.2.183- the newest at this time of writing was not working)
4. Restart the project at this point in order to regenerate the libs.
5. In file explorer delete the Library folder in your project
6. Open in Unity with your platform selected (Android) as it gets reseted after you delete the library folder
7. At this point External Dependency Man -> Force Resolve
8. Build

Hope this helps to anyone.
